tsl/src/compression/compression.c

```diff
@@ -161,6 +161,49 @@ truncate_relation(Oid table_oid)
 	table_close(rel, NoLock);
 }
 
+/* Handle the all rows deletion of a given relation */
+static void
+RelationDeleteAllRows(Relation rel, Snapshot snap)
+{
+	TupleTableSlot *slot = table_slot_create(rel, NULL);
+	TableScanDesc scan = table_beginscan(rel, snap, 0, (ScanKey) NULL);
+
+	while (table_scan_getnextslot(scan, ForwardScanDirection, slot))
+	{
+		bool should_free;
+		HeapTuple tuple = ExecFetchSlotHeapTuple(slot, false, &should_free);
+
+		simple_table_tuple_delete(rel, &tuple->t_self, snap);
+	}
+	table_endscan(scan);
+	ExecDropSingleTupleTableSlot(slot);
+}
+
+/*
+ * Delete the relation WITHOUT applying triggers. This will be used when
+ * `enable_delete_after_compression = true` instead of truncating the relation.
+ * Also don't restart sequences.
+ */
+static void
+delete_relation_rows(Oid table_oid)
+{
+	Relation rel = table_open(table_oid, RowExclusiveLock);
+	Snapshot snap = GetActiveSnapshot();
```

Comment:
I'm a little paranoid about this since we use GetLatestSnapshot to fetch the tuples for compression. These may very well end up being different snapshots (at least the active transaction list could be different) and delete different set of tuples then that were compressed.

I'd rather use the same STATIC snapshot for this by keeping a reference to it. Better safe than sorry.

Reply:
Make sense, changed to use `GetLatestSnapshot` instead